pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -117,6 +116,10 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 				pc.deleteIP(ip)
 			}
 		}
+		pc.Unlock()
+		if pc.c.serviceEntryCallback != nil {
```

Comment:
nit: I don't think this should be called `serviceEntryCallback`. The pod conrtoller doesn't care about ServiceEntries, it just triggers some opaque callback when a pod is changed. Maybe `podUpdateCallback` or `podCallback` or `podEventCallback` or `podEventHandler`?